File: Harness Typescript/src/evaluators/exact-match.ts
import { BaseEvaluator, EvaluatorResult } from './base.js';

export class ExactMatchEvaluator extends BaseEvaluator {
  constructor() {
    super("exact_match");
  }

  async evaluate(output: string, expected: any): Promise<EvaluatorResult> {
    const passed = output.trim() === String(expected).trim();
    if (!passed) {
      const displayOut = output.length > 50 ? output.substring(0, 50) + "..." : output;
      return {
        passed: false,
        score: 0.0,
        reason: `Expected '${expected}', got '${displayOut}'`
      };
    }
    return { passed: true, score: 1.0 };
  }
}

File: Harness Python/evaluators/exact_match.py
from evaluators.base import BaseEvaluator, EvalResult

class ExactMatchEvaluator(BaseEvaluator):
    async def evaluate(self, output: str, test_case: dict) -> EvalResult:
        expected = test_case.get("expected", "").strip().lower()
        actual = output.strip().lower()
        
        if actual == expected:
            return EvalResult(passed=True, score=1.0)
        return EvalResult(passed=False, score=0.0, reason=f"Expected '{expected}', got '{actual}'")
